File: packages/flutter-dart/lib/appkit_config/cinacoin_theme.dart
import 'package:flutter/material.dart';

/// Cinacoin brand theme configuration
class CinacoinTheme {
  final Color primary;
  final Color secondary;
  final Color background;
  final Color surface;
  final Color text;
  final Color textSecondary;
  final Color border;
  final Color error;
  final Color success;

  final CinacoinTypography typography;
  final CinacoinSpacing spacing;
  final CinacoinBorderRadius borderRadius;

  const CinacoinTheme({
    required this.primary,
    required this.secondary,
    required this.background,
    required this.surface,
    required this.text,
    required this.textSecondary,
    required this.border,
    required this.error,
    required this.success,
    this.typography = const CinacoinTypography(),
    this.spacing = const CinacoinSpacing(),
    this.borderRadius = const CinacoinBorderRadius(),
  });

  /// Dark theme (default)
  static const dark = CinacoinTheme(
    primary: Color(0xFF0070F3),
    secondary: Color(0xFF6366F1),
    background: Color(0xFF000000),
    surface: Color(0xFF141414),
    text: Color(0xFFFFFFFF),
    textSecondary: Color(0xFFA1A1AA),
    border: Color(0xFF27272A),
    error: Color(0xFFEF4444),
    success: Color(0xFF22C55E),
  );

  /// Light theme
  static const light = CinacoinTheme(
    primary: Color(0xFF0070F3),
    secondary: Color(0xFF6366F1),
    background: Color(0xFFFFFFFF),
    surface: Color(0xFFF4F4F5),
    text: Color(0xFF000000),
    textSecondary: Color(0xFF71717A),
    border: Color(0xFFE4E4E7),
    error: Color(0xFFEF4444),
    success: Color(0xFF22C55E),
  );
}

/// Typography configuration
class CinacoinTypography {
  final double small;
  final double medium;
  final double large;
  final double xlarge;

  const CinacoinTypography({
    this.small = 12,
    this.medium = 14,
    this.large = 16,
    this.xlarge = 20,
  });
}

/// Spacing configuration
class CinacoinSpacing {
  final double xs;
  final double sm;
  final double md;
  final double lg;
  final double xl;

  const CinacoinSpacing({
    this.xs = 4,
    this.sm = 8,
    this.md = 16,
    this.lg = 24,
    this.xl = 32,
  });
}

/// Border radius configuration
class CinacoinBorderRadius {
  final double small;
  final double medium;
  final double large;

  const CinacoinBorderRadius({
    this.small = 4,
    this.medium = 8,
    this.large = 12,
  });
}

File: packages/android-kotlin/src/main/kotlin/com/cinacoin/appkit/config/CinacoinTheme.kt
package com.cinacoin.appkit.config

import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.TextUnit
import androidx.compose.ui.unit.sp

/**
 * Cinacoin brand theme configuration
 */
data class CinacoinTheme(
    val primary: Color,
    val secondary: Color,
    val background: Color,
    val surface: Color,
    val text: Color,
    val textSecondary: Color,
    val border: Color,
    val error: Color,
    val success: Color,
    val typography: Typography = Typography.Default,
    val spacing: Spacing = Spacing.Default,
    val borderRadius: BorderRadius = BorderRadius.Default
) {
    data class Typography(
        val small: TextUnit,
        val medium: TextUnit,
        val large: TextUnit,
        val xlarge: TextUnit
    ) {
        companion object {
            val Default = Typography(
                small = 12.sp,
                medium = 14.sp,
                large = 16.sp,
                xlarge = 20.sp
            )
        }
    }
    
    data class Spacing(
        val xs: androidx.compose.ui.unit.Dp,
        val sm: androidx.compose.ui.unit.Dp,
        val md: androidx.compose.ui.unit.Dp,
        val lg: androidx.compose.ui.unit.Dp,
        val xl: androidx.compose.ui.unit.Dp
    ) {
        companion object {
            val Default = Spacing(
                xs = 4.dp,
                sm = 8.dp,
                md = 16.dp,
                lg = 24.dp,
                xl = 32.dp
            )
        }
    }
    
    data class BorderRadius(
        val small: androidx.compose.ui.unit.Dp,
        val medium: androidx.compose.ui.unit.Dp,
        val large: androidx.compose.ui.unit.Dp
    ) {
        companion object {
            val Default = BorderRadius(
                small = 4.dp,
                medium = 8.dp,
                large = 12.dp
            )
        }
    }
    
    companion object {
        val Dark = CinacoinTheme(
            primary = Color(0xFF0070F3),
            secondary = Color(0xFF6366F1),
            background = Color(0xFF000000),
            surface = Color(0xFF141414),
            text = Color(0xFFFFFFFF),
            textSecondary = Color(0xFFA1A1AA),
            border = Color(0xFF27272A),
            error = Color(0xFFEF4444),
            success = Color(0xFF22C55E)
        )
        
        val Light = CinacoinTheme(
            primary = Color(0xFF0070F3),
            secondary = Color(0xFF6366F1),
            background = Color(0xFFFFFFFF),
            surface = Color(0xFFF4F4F5),
            text = Color(0xFF000000),
            textSecondary = Color(0xFF71717A),
            border = Color(0xFFE4E4E7),
            error = Color(0xFFEF4444),
            success = Color(0xFF22C55E)
        )
    }
}
